### PR TITLE
DR-1112: Evaluate and update npm packages

### PR DESCRIPTION
**Pull request by AITR1**
## Description

Update npm packages so that have the latest compatible versions.

## Context

Update outdated packages
- Major version bumps if necessary (not nhsuk-frontend as nhsuk-react-components doesn't support v9)
- Minor package bumps for consistency

Remove CRA required several additional changes
- Implementation of vite (moving and updating of index.html, vite config file, update of start and build scripts, change process.env to import metal env)
- Implementation of jest (updating jest config, adding jest-dom env package, updating failing tests due to types, removing warnings in test runs, remove jest-mock-fetch, update test-setup file, add mock for import.meta.env)

## Type of changes

- [X] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [X] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [X] I have followed the code style of the project
- [X] I have added tests to cover my changes
- [] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [X] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: src/frontend/src/pages/organisations/__tests__/add.test.tsx [hunk 1]
```diff
@@ -1,10 +1,11 @@
-import { act, render, screen, waitFor } from '@testing-library/react';
+import { render, screen, waitFor } from '@testing-library/react';
+import { act } from 'react';
 import AddOrganisationPage from '../add';
 
 beforeEach(async () => {
   await act(async () => {
     render(<AddOrganisationPage />);
-  });
+});
```
Comment (AITR1) [inline src/frontend/src/pages/organisations/__tests__/add.test.tsx:8]: indent

File: src/frontend/src/pages/organisations/__tests__/search.test.tsx [hunk 2]
```diff
@@ -34,47 +34,43 @@ describe('SearchOrganisationPage', () => {
...
+    const nextButton = screen.getByTestId('next-button');
+    await userEvent.click(nextButton);
 
     await waitFor(() => {
-      const nextButton = screen.getByTestId('next-button');
-      userEvent.click(nextButton);
+      expect(navigateMock).toHaveBeenCalledWith('/organisations/add');
     });
 
     expect(navigateMock).toHaveBeenCalledWith('/organisations/add');
```
Comment (AITR1) [inline src/frontend/src/pages/organisations/__tests__/search.test.tsx:76]: remove

File: src/frontend/src/components/cardGroup/__tests__/cardGroup.test.tsx [hunk 3]
```diff
@@ -8,7 +9,7 @@ describe('CardGroup', () => {
       { href: 'link1', text: 'text1', description: 'description1', dataTestId: "datatestid1" },
       { href: 'link2', text: 'text2', description: 'description2', dataTestId: "datatestid2" },
       { href: 'link3', text: 'text3', description: 'description3', dataTestId: "datatestid3" },
-    ];
+    ] as unknown as [CardInfo];
```
Comment (Tomdango) [inline src/frontend/src/components/cardGroup/__tests__/cardGroup.test.tsx:12]: You likely need to cast this to unknown first as the typing should be `CardInfo[]` to represent an array of 0 or more CardInfo objects. The current typing implies an array that always has only one CardInfo object

File: src/frontend/src/services/api/__tests__/controller.test.ts [hunk 4]
```diff
@@ -30,16 +30,23 @@ describe('getApi', () => {
...
 
-  it('should return RemoteMockApi when REACT_APP_API_MODE is "REMOTEMOCK"', () => {
-    process.env.REACT_APP_API_MODE = 'REMOTEMOCK';
+  it('should return RemoteMockApi when VITE_APP_API_MODE is "REMOTEMOCK"', () => {
+    jest.resetModules();
+    jest.mock('../../../config/config', () => ({
+      config: {
+        VITE_APP_API_MODE: 'REMOTEMOCK',
+      },
+    }));
```
Comment (Tomdango) [inline src/frontend/src/services/api/__tests__/controller.test.ts:44]: Does this impact the global mock in `jest-setup.ts`?

File: src/frontend/src/styles/global.css [hunk 5]
```diff
@@ -2,10 +2,3 @@
     background: #007f3b;
     color: #ffffff;
 }
-
-.app-footer {
-    background-color: #d8dde0;
-    border-top: 4px solid #005eb8;
-    padding-bottom: 24px;
-    padding-top: 24px
-}
```
Comment (Tomdango) [inline src/frontend/src/styles/global.css:11]: I guess this answers my above question r.e. the footer. But how come it's removed in this ticket?

**Review by Tomdango — COMMENTED** [review]
Looks good - just a few small comments then good to approve 👍

File: src/frontend/src/App.tsx [hunk 6]
```diff
@@ -30,7 +30,7 @@ function App() {
           <Route path="*" element={<OrganisationsPage />} />
         </Routes>
       </div>
-      <div className="app-footer">
```
[the remark this answers is not in the document]
Comment (AITR1) [reply src/frontend/src/App.tsx:33]: When upgrading to the latest version of the react nhs component package it resulted in a double line in the footer, removing this class reverted it to only use the styling from the component.

File: src/frontend/src/services/api/__tests__/controller.test.ts (hunk 4 above)
Comment (AITR1) [reply src/frontend/src/services/api/__tests__/controller.test.ts:44]: Yes, the global mock set it as 'LOCALMOCK' however in this instance we needed to test it as 'REMOTEMOCK' - after a few work arounds this is the only I got round it was calling the api again after mocking the config in the test

File: src/frontend/src/styles/global.css (hunk 5 above)
Comment (AITR1) [reply src/frontend/src/styles/global.css:11]: See above, no longer needed. Styling looks as intended without it

File: src/frontend/src/components/cardGroup/__tests__/cardGroup.test.tsx (hunk 3 above)
Comment (AITR1) [reply src/frontend/src/components/cardGroup/__tests__/cardGroup.test.tsx]: Updated to be as suggested

**Review by Tomdango — APPROVED** [review]
LGTM 👍

**Pull request merged**